framework/auth/campaigns.py

```diff
@@ -75,24 +75,49 @@ def get_campaigns():
                     }
                 })
 
+            # Collections
+            collection_providers = CollectionProvider.objects.all()
+            for collection_provider in collection_providers:
+                campaign = f'{collection_provider._id}-collections'
+                system_tag = provider_source_tag(collection_provider._id, 'collection')
+                name = collection_provider.name
+                url_path = f'collections/{collection_provider._id}'
+                newest_campaigns.update({
+                    campaign: {
+                        'system_tag': system_tag,
+                        'redirect_url': furl.furl(DOMAIN).add(path=url_path),
+                        'confirmation_email_template': mails.CONFIRM_EMAIL_COLLECTIONS(name),
```

Comment:
Verifying that people only get one email to join the OSF even if they would otherwise get several claim tags. Also, what wins the email branding when there's multiple possible templates?

Reply:
The campaigns workflow applies to the application of a single source tag depending on the origin of the 'sign-up' button click. It's based off the `campaign` or `destination` (if external provider login) qsarg so only one campaign, and its respective email template, will be chosen from the `get_campaigns` dictionary.